lib/schedule.ts

```diff
@@ -24,25 +24,48 @@ export class Schedule {
     cron: string,
     callback: JobCallback,
     config?: ICronJobConfig,
+    tryLock?: Promise<TryLock> | TryLock,
```

Comment:
Dynamic job is not support distribution now, because the job is not sync to other nodes in cluster, tryLock function in this has no effect. Maybe I will resolve it in the future.

Reply:
I think it still ok for some cases. In my project, user can upload csv files. My app (which is running in multiple instances) will try to pick up the csv file to process data. In this case, I have to make sure that the mutex lock is good enough to not let other instances pick up the same file.